src/library/scala/collection/mutable/RedBlackTree.scala

```diff
@@ -0,0 +1,502 @@
+package scala.collection.mutable
+
+import scala.annotation.tailrec
+import scala.collection.Iterator
+
+/**
+ * An object containing the red-black tree implementation used by mutable `TreeMaps`. For performance reasons, this
+ * implementation uses `null` references to represent leaves instead of a sentinel node. Currently, the internal nodes
+ * do not store their subtree size - only the tree object keeps track of their size. Therefore, while obtaining the size
+ * of the whole tree is O(1), knowing the number of entries inside a range is O(n) on the size of the range.
```

Comment:
I'd say it's worthwhile to leave the complexity comments in the user API docs, and move everything else into a normal comment. Perhaps you could alternatively move the O(n) comment about the range size to the respective method.